code/controllers/subsystem/pai.dm

```diff
@@ -3,198 +3,168 @@ SUBSYSTEM_DEF(pai)
+	var/request_spam = FALSE
+	/// Prevents a pAI from submitting itself repeatedly and sounding an alert.
+	var/submit_spam = FALSE
+	/// All pAI cards on the map.
 	var/list/pai_card_list = list()
 
-/datum/controller/subsystem/pai/Topic(href, href_list[])
-	if(href_list["download"])
-		var/datum/paiCandidate/candidate = locate(href_list["candidate"]) in candidates
-		var/obj/item/paicard/card = locate(href_list["device"]) in pai_card_list
-		if(card.pai)
-			return
-		if(istype(card, /obj/item/paicard) && istype(candidate, /datum/paiCandidate))
-			if(check_ready(candidate) != candidate)
-				return FALSE
-			var/mob/living/silicon/pai/pai = new(card)
-			if(!candidate.name)
-				pai.name = pick(GLOB.ninja_names)
-			else
-				pai.name = candidate.name
-			pai.real_name = pai.name
-			pai.key = candidate.key
-
-			card.setPersonality(pai)
-
-			SSticker.mode.update_cult_icons_removed(card.pai.mind)
-
-			candidates -= candidate
-			usr << browse(null, "window=findPai")
-
-	if(href_list["new"])
-		var/datum/paiCandidate/candidate = locate(href_list["candidate"]) in candidates
-		var/option = href_list["option"]
-		var/t = ""
-
-		switch(option)
-			if("name")
-				t = sanitize_name(stripped_input(usr, "Enter a name for your pAI", "pAI Name", candidate.name, MAX_NAME_LEN))
-				if(t)
-					candidate.name = t
-			if("desc")
-				t = stripped_multiline_input(usr, "Enter a description for your pAI", "pAI Description", candidate.description)
-				if(t)
-					candidate.description = t
-			if("role")
-				t = stripped_input(usr, "Enter a role for your pAI", "pAI Role", candidate.role)
-				if(t)
-					candidate.role = t
-			if("ooc")
-				t = stripped_multiline_input(usr, "Enter any OOC comments", "pAI OOC Comments", candidate.comments)
-				if(t)
-					candidate.comments = t
-			if("save")
-				candidate.savefile_save(usr)
-			if("load")
-				candidate.savefile_load(usr)
-				//In case people have saved unsanitized stuff.
-				if(candidate.name)
-					candidate.name = copytext_char(sanitize(candidate.name),1,MAX_NAME_LEN)
-				if(candidate.description)
-					candidate.description = copytext_char(sanitize(candidate.description),1,MAX_MESSAGE_LEN)
-				if(candidate.role)
-					candidate.role = copytext_char(sanitize(candidate.role),1,MAX_MESSAGE_LEN)
-				if(candidate.comments)
-					candidate.comments = copytext_char(sanitize(candidate.comments),1,MAX_MESSAGE_LEN)
-
-			if("submit")
-				if(candidate)
-					candidate.ready = 1
-					for(var/obj/item/paicard/p in pai_card_list)
-						if(!p.pai)
-							p.alertUpdate()
-				usr << browse(null, "window=paiRecruit")
-				return
-		recruitWindow(usr)
-
-/datum/controller/subsystem/pai/proc/recruitWindow(mob/M)
-	var/datum/paiCandidate/candidate
-	for(var/datum/paiCandidate/c in candidates)
-		if(c.key == M.key)
-			candidate = c
-	if(!candidate)
-		candidate = new /datum/paiCandidate()
-		candidate.key = M.key
+/// Created when a user clicks the "pAI candidate" window
+/datum/pai_candidate
```

Comment:
I'm not sure if these are filtered, but all of them (minus `key` & `ready`) need to be chat filtered when they're inputted.